tests/test_cleaning.py

```diff
@@ -0,0 +1,234 @@
+from unittest.mock import Mock, patch
+import pytest
+from typing import List, Generator
+from personal_finances.transaction.cleaning import remove_internal_transfers
+from personal_finances.transaction.definition import SimpleTransaction
+from datetime import datetime
+
+
+@pytest.fixture(autouse=True)
+def user_config_mock() -> Generator[Mock, None, None]:
+    with patch(
+        "personal_finances.transaction.cleaning.get_user_configuration"
+    ) as u_mock:
+        u_mock.return_value.InternalTransferReferences = ["internal_reference_1"]
+        u_mock.return_value.BankProcessingTimeInDays = 5
+        yield u_mock
+
+
+@pytest.mark.parametrize(
+    "raw_transactions,clean_transactions",
+    [
```

Comment:
That's a long block of code inside this list, it's hard to make sense of each test without careful reading. We can make it more readable by giving test cases a name and referencing here.

Transaction lists that look exactly the same can also be named and reused to make reasoning clearer and less error prone.

```
NO_INTERNAL_TRANSFER_LIST = [...]

EMPTY_TRANSACTIONS_TEST = ([], [])
NO_INTERNAL_TRANSFER_TEST = (
    NO_INTERNAL_TRANSFER_LIST,
    NO_INTERNAL_TRANSFER_LIST
)
ONE_INTERNAL_TRANSFER_TEST = ([...], [...])
...

@pytest.mark.parametrize(
    "raw_transactions,clean_transactions",
    [
        EMPTY_TRANSACTIONS,
        NO_INTERNAL_TRANSFER,
        ...
    ]
)
```

Reply:
Alternatively, you can implement builder functions to handle more complex cases.

Reply:
For example:

```python
def add_internal_transfer(transaction_list: List[SimpleTransaction]):
    transfer_amount = randn()
    transaction_list.add(SimpleTransaction(.., amount=amount))
    transaction_list.add(SimpleTransaction(.., amount=-amount))
```